wallet-new/src/Cardano/Wallet/Kernel/CoinSelection/FromGeneric.hs

```diff
@@ -408,4 +444,53 @@ estimateSize saa sta ins outs
 --         here with some (hopefully) realistic values.
 estimateCardanoFee :: TxSizeLinear -> Int -> [Word64] -> Word64
 estimateCardanoFee linearFeePolicy ins outs
-    = round (calculateTxSizeLinear linearFeePolicy (estimateSize 128 16 ins outs))
+    = round $ calculateTxSizeLinear linearFeePolicy
+            $ estimateSize dummyAddrAttrSize dummyTxAttrSize ins outs
+
+-- | Size to use for a value of type @Attributes AddrAttributes@ when estimating
+--   encoded transaction sizes. The minimum possible value is 2.
+dummyAddrAttrSize :: Byte
+dummyAddrAttrSize = 16
+
+-- | Size to use for a value of type @Attributes ()@ when estimating
+--   encoded transaction sizes. The minimum possible value is 2.
+dummyTxAttrSize :: Byte
+dummyTxAttrSize = 2
+
+-- | For a given transaction size, and sizes for @Attributes AddrAttributes@ and
+--   @Attributes ()@, compute the maximum possible number of inputs a transaction
+--   can have. The formula for this value is not linear, so we just do a binary
+--   search here. A decent first guess makes this search relatively quick.
+--
+--   We use a conservative over-estimate for the encoded transaction sizes, so the
+--   number of transaction inputs computed here is a lower bound on the true
+--   maximum.
+estimateMaxTxInputs
+  :: Byte -- ^ Size of @Attributes AddrAttributes@
+  -> Byte -- ^ Size of @Attributes ()@
+  -> Byte -- ^ Maximum size of a transaction
+  -> Word64
+estimateMaxTxInputs addrAttrSize txAttrSize maxSize = fromIntegral $
```

Comment:
Do you think it's possible to write trivial unit tests for this function? It's used in a crucial part of the new wallet and is not a one-liner, which might suggest testing it out is a good thing to do 😉 

Also, minor bikeshed -- I'm wondering whether we should rehome these calculations into a more specific module, leaving `FromGeneric` free from this "fee internals", and re-export in this module only the top-level functions.

I would also curious to hear what @edsko thinks 😉